Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -1336,6 +1392,110 @@ bool IsTracking() const {
+                };
+
+                ARReferenceImage* referenceImage{[[ARReferenceImage alloc]
+                    initWithCGImage: image
+                    orientation: CGImagePropertyOrientation::kCGImagePropertyOrientationUp
+                    physicalWidth: request.measuredWidthInMeters == 0 ? 1 : request.measuredWidthInMeters]};
+                
+                // Store the index in the name field.
+                referenceImage.name = [NSString stringWithFormat:@"%zu", i];
+                
+                // Queue image validation
+                if (@available(iOS 13.0, *)) {
+                    dispatch_group_enter(validationGroup);
+                    dispatch_async(validationQueue, ^{
+                        [referenceImage validateWithCompletionHandler:^(NSError * _Nullable error) {
+                            if (error != nil) {
+                                imageTrackingScores[i] = ImageTrackingScore::UNTRACKABLE;
+                            } else {
+                                imageTrackingScores[i] = ImageTrackingScore::TRACKABLE;
+                                
+                                // Add image to our image set if it is trackable.
+                                [imageSet addObject: referenceImage];
+                                imageCount++;
+                            }
+                            
+                            dispatch_group_leave(validationGroup);
+                        }];
+                    });
+                } else {
+                    imageTrackingScores[i] = ImageTrackingScore::TRACKABLE;
+                }
+                
+                CGImageRelease(image);
+                CGDataProviderRelease(provider);
+                CGColorSpaceRelease(colorSpace);
+            }
+                
+            // Wait for all scores to calculated on a separate scheduler.
+            dispatch_queue_t awaiterQueue{dispatch_queue_create("Awaiter Queue", nil)};
+            dispatch_async(awaiterQueue, ^{
+                dispatch_group_wait(validationGroup, DISPATCH_TIME_FOREVER);
+
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    // Convert each image request from a bitmap to a CGImage, and prepare to pass that to the ARKit configuration.
+                    ARWorldTrackingConfiguration* configuration{static_cast<ARWorldTrackingConfiguration*>(SystemImpl.XrContext->Session.configuration)};
+
+                    // If we have any images that qualified for tracking then enable image detection.
+                    if (imageCount > 0 && configuration != nil) {
+                        configuration.detectionImages = imageSet;
+                        
+                        if (@available(iOS 13.0, *)) {
+                            configuration.automaticImageScaleEstimationEnabled = true;
+                        }
+
+                        configuration.maximumNumberOfTrackedImages = imageCount > 4 ? 4 : imageCount;
```

Comment:
Are images beyond 4 just ignored then?

Reply:
In ARKit there are basically "high quality" tracked images which can be updated every frame which is what this max number represents.  If any of those images leaves the viewport then it will free up room to start tracking additional images.

If you leave this setting at the default of 0 then image tracking does not enforce this limit, but anchors are updated very slowly at ~1 tick per second.

https://developer.apple.com/documentation/arkit/arworldtrackingconfiguration/2968182-maximumnumberoftrackedimages